detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeRecipesParser.java

```diff
@@ -78,4 +67,13 @@ private BitbakeRecipe parseLine(String line, BitbakeRecipe currentRecipe, Map<St
             return currentRecipe;
         }
     }
+
+    private void finishCurrentRecipe(final Map<String, List<String>> bitbakeRecipes, final Set<String> layerNames, final BitbakeRecipe currentRecipe) {
+        if (currentRecipe != null) {
+            bitbakeRecipes.put(currentRecipe.getName(), currentRecipe.getLayerNames());
+            if (currentRecipe.getLayerNames() != null) {
+                layerNames.addAll(currentRecipe.getLayerNames());
+            }
+        }
```

Comment:
Is any logging warranted? I'm confused as to why the current recipe could be `null`.

Reply:
I actually haven't changed that parser behavior, only refactored it so it's now in a different place. As far as I know the parser is working correctly.